### ISSUE TITLE
Support ChatGPT work with Apps on macOS feature

### ISSUE THREAD

**Issue by markusmoenig** [zed-industries/zed#21835]
### Check for existing issues

- [X] Completed

### Describe the feature

Support

https://help.openai.com/en/articles/10119604-work-with-apps-on-macos

Xcode / TextEditor on the Mac all do this already.

### Environment

Zed: v0.164.2 (Zed)
OS: macOS 15.1.1
Memory: 24 GiB
Architecture: aarch64

**Comment by mielsense** [comment]
bump

**Comment by balboah** [comment]
This is very important for me coming from vscode, it's very convenient to quickly ask chatgpt something across different apps (or even multiple at a time). Also not having to spend AI request quotas for simple things.

Maybe it can be implemented like an extension, like they did for vscode?
Not sure if it needs to be approved by ChatGPT or Zed team in some way to get it to show up as compatible app

**Comment by markusmoenig** [comment]
I don't think we will get this. It will be a big amount of work to create a wrapper around their API which can be served to the system and they earn their money with integrating LLMs directly into the editor. So why should they spend a large amount of work on something which competes with their own monetization.

**Comment by balboah** [comment]
This only allows editing/viewing one file at a time, I don't think it competes with the agentic AI features (except maybe the inline edit popup). If it can be done as an extension, anyone can try to do it. But it seems to me it's not possible currently as a 3rd party, I think openai team has to do it.

From their support bot:
> More integrations may be available in the future, but there isn’t a public submission process to directly request or add a new app at this time

So perhaps this task is to persuade openai team to start using Zed as their default :P
